src/app/components/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -9,19 +11,65 @@ import { Subscription } from 'rxjs';
 })
 export class BreadcrumbsComponent implements OnInit, OnDestroy {
   public isAuthenticated: boolean;
+  public breadcrumbs: IBreadCrumb[];
 
   private subscriptions = new Subscription();
 
-  constructor(private authService: AuthService) {}
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+    private activatedRoute: ActivatedRoute
+  ) {}
 
   public ngOnInit(): void {
+
+    this.breadcrumbs = this.buildBreadCrumb(this.activatedRoute.root);
     this.isAuthenticated = this.authService.isAuthenticated();
     this.subscriptions.add(this.authService.isAuth$.subscribe((isAuth) => {
-      this.isAuthenticated = isAuth
+      this.isAuthenticated = isAuth;
+    }));
+
+    this.subscriptions.add(this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
+        this.breadcrumbs = this.buildBreadCrumb(this.activatedRoute.root);
```

Comment:
1. You could use activatedRoute.url to understand which page is opened now. 
2. Also you could keep contants for labels for breadcrumbs in a separate file instad of passing it via routing. 

There is no need now to refactor it somehow, but maybe points above could simplify the logic of buildBreadCrumb method. Just take into account for the future.

Reply:
Ok. Thank you.